src/Network/Ethereum/Core/BigNumber.purs

```diff
@@ -119,13 +121,25 @@ unsafeToInt = Int.floor <<< toNumber
 -- | Take the integer part of a big number
 foreign import floorBigNumber :: BigNumber -> BigNumber
 
-foreign import toBigNumber :: Foreign -> BigNumber
-
 instance decodeBigNumber :: Decode BigNumber where
-  decode = pure <<< toBigNumber
+  decode a = do
+    str <- readString a
+    case parseBigNumber Int.hexadecimal str of
+      Nothing -> fail <<< ForeignError $ "Could not parse BigNumber from stringified Integer: " <> str
+      Just res -> pure res
 
 instance readFBigNumber :: ReadForeign BigNumber where
   readImpl = decode
 
 instance encodeBigNumber :: Encode BigNumber where
   encode = encode <<< (append "0x") <<< toString Int.hexadecimal
+
+instance decodeJsonBigNumber :: A.DecodeJson BigNumber where
+  decodeJson json = do
+    str <- A.decodeJson json
```

Comment:
Maybe extract common part from `decodeJson` and `decode` in:
`decode_ :: forall m. MonadThrow String m => String -> m BigNumber`  or something like that (I think Either could work instead of MonadThrow too)